lib/build-cloud.rb

```diff
@@ -220,16 +223,31 @@ def recursive_interpolate_config(h)
         when String
 
             while h =~ /%\{(.+?)\}/
```

Comment:
`while /^%\{(?<var>[^\|]*)(?:\|{2}(?<default>.*))?\}$/`  Should automatically give you `var` and `default` within the `while` loop. If no default was given, `default` will be `nil`.